### PR TITLE
Remove MacOS-specific "generation"

### PR DESCRIPTION
**Pull request by eeeebbbbrrrr**
This is an initial stab at removing the MacOS-specific "generation" stuff and instead using the `pg_proc` entry's "xmin" and "cmin" values to formulate one that's based on the tuple's immutable characteristics.  And this does it for all platforms.

We also now make a distinction between the "crate_name" which includes this number and the "symbol name" which does not.  We store the symbol name as an `Option<String>` in our json value in `prosrc` and will use a backwards compatible default for Linux when loading a user function that doesn't have the `"symbol"` property.  No concern was given to being backwards compatible for any existing MacOS plrust functions because... MacOS.

### CODE REVIEW

File: plrust/src/pgproc.rs [hunk 1]
```diff
@@ -76,6 +76,28 @@ impl PgProc {
...
+                .unwrap_unchecked() // SAFETY: t_data will never be null and `cmin()` is called in a potentially hot path
+                .t_choice
+                .t_heap
+                .t_field3
+                .t_cid
+        }
+    }
+
+    #[inline]
+    pub(crate) fn generation_number(&self) -> u64 {
```
Comment (eeeebbbbrrrr) [inline plrust/src/pgproc.rs:97]: Couldn't decide if we wanted a stronger type than `u64`.  I considered `struct GenerationNumber { xmin: TransactionId, cmin: CommandId }` but didn't want to do all that typing if this idea turns out bad.

Comment (workingjubilee) [reply plrust/src/pgproc.rs:97]: A u64 is fine for an incrementing counter, we can lock things down later if needs be.

File: plrust/src/plrust.rs [hunk 2]
```diff
@@ -136,7 +136,15 @@ pub(crate) fn compile_function(fn_oid: pg_sys::Oid) -> eyre::Result<Output> {
     Ok(this_output.unwrap())
 }
 
-pub(crate) fn crate_name(db_oid: pg_sys::Oid, fn_oid: pg_sys::Oid) -> String {
+pub(crate) fn symbol_name(db_oid: pg_sys::Oid, fn_oid: pg_sys::Oid) -> String {
+    format!("plrust_fn_oid_{}_{}", db_oid.as_u32(), fn_oid.as_u32())
+}
+
+pub(crate) fn crate_name(
```
Comment (workingjubilee) [inline plrust/src/plrust.rs:143]: Maybe these should explain when to use the different names?

**Review by workingjubilee — APPROVED** [review]
This looks good to me!

File: plrust/src/plrust.rs (hunk 2 above)
Comment (eeeebbbbrrrr) [reply plrust/src/plrust.rs:143]: sure, I'll take care of that.

**Pull request merged**